### ISSUE TITLE
[a11y]: ContainedList [with expandable search] - need clarity

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13942]
### Package

@carbon/react

### Package version

carbon/react@1.30.0

### Automated testing tool and ruleset

n/a

### Assistive technology

all

### Description

Search button

Search button is still available even after the search field is displayed

The search button is still available after the search edit field is displayed. however it does not perform any action upon user activation. 

### WCAG 2.1 Violation

n/a

### Reproduction/example

Record `Number 30` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

### Steps to reproduce

Recommended fix: If the search button does not perform any action, it can be disabled.

This is an observation

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
Need to add `aria-hidden={expanded ? false : true}` to `ExpandableSearch`

**Comment by mateus037** [comment]
Hi @sstrubberg @tay1orjones Could I work on this issue? :D

**Comment by sstrubberg** [comment]
> Hi [@sstrubberg](https://github.com/sstrubberg) [@tay1orjones](https://github.com/tay1orjones) Could I work on this issue? :D
✅
